dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -6329,12 +6234,216 @@ else if(field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())){
+                    .HOST_FOLDER_FIELD)).asOldField());
+            throw cve;
+        }
+
+        //Enforce validation only if the file name isn't the same we've already got
+        if(hasNewIncomingFile(contentlet)){
+            boolean fileNameExists = false;
+            try {
+                final Host site = APILocator.getHostAPI ().find(contentlet.getHost(), APILocator.getUserAPI().getSystemUser(), false);
+                final Folder folder = UtilMethods.isSet(contentlet.getFolder())?
+                        APILocator.getFolderAPI().find(contentlet.getFolder(), APILocator.getUserAPI().getSystemUser(), false):
+                        APILocator.getFolderAPI().findSystemFolder();
+
+                String fileName = contentlet.getBinary(FileAssetAPI.BINARY_FIELD) != null ? contentlet.getBinary(FileAssetAPI.BINARY_FIELD).getName() : StringPool.BLANK;
+                if(UtilMethods.isSet(contentlet.getStringProperty("fileName"))){
+                    fileName = contentlet.getStringProperty("fileName");
+                }
+                if(UtilMethods.isSet(fileName)){
+                    fileNameExists = APILocator.getFileAssetAPI().fileNameExists(site, folder, fileName, contentlet.getIdentifier());
+                    if(!APILocator.getFolderAPI().matchFilter(folder, fileName)) {
+                        final DotContentletValidationException cve = new FileAssetValidationException("message.file_asset.error.filename.filters");
+                        Logger.warn(this, "File Asset [" + contentIdentifier + "] does not match specified folder" +
+                                " file filters");
+                        cve.addBadTypeField(new LegacyFieldTransformer(contentType.fieldMap().get(FileAssetAPI
+                                .HOST_FOLDER_FIELD)).asOldField());
+                        throw cve;
+                    }
+                }
+
+            } catch (final Exception e) {
+                if(e instanceof FileAssetValidationException) {
+                    throw (FileAssetValidationException) e;
+                }
+                final String errorMsg = "Unable to validate field: " + FileAssetAPI.BINARY_FIELD + " in " +
+                        "contentlet [" + contentIdentifier + "]";
+                Logger.warn(this, errorMsg);
+                throw new FileAssetValidationException(errorMsg, e);
+            }
+            if(fileNameExists){
+                final DotContentletValidationException cve = new FileAssetValidationException("message.contentlet.fileasset.filename.already.exists");
+                Logger.warn(this, "Name of File Asset [" + contentIdentifier + "] already exists");
+                cve.addBadTypeField(new LegacyFieldTransformer(contentType.fieldMap().get(FileAssetAPI
+                        .HOST_FOLDER_FIELD)).asOldField());
+                throw cve;
+            }
+        }
+    }
+
```

Comment:
maybe we can move all this validateXXX method for a Util class or create a Validator class for each field type